### PR TITLE
Feature/main/246 scrape pillowfort

### PR DESCRIPTION
**Pull request by Erhannis**
### Before you begin

* I understand my contributions may be rejected for any reason
* I understand my contributions are for the benefit of Derpibooru and/or the Philomena software
* I understand my contributions are licensed under the GNU AGPLv3

- [X] I understand all of the above

---

Added support for scraping Pillowfort urls.  I wasn't super sure about how robust we wanted to be to changes to Pillowfort's url, but it seems to work for now, anyway.

### CODE REVIEW

**Comment by Erhannis** [comment]
Oh, I should comment explicitly that this is to resolve https://github.com/derpibooru/philomena/issues/246

File: lib/philomena/scrapers/pillowfort.ex [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  defp json!({:ok, %Tesla.Env{body: body, status: 200}}),
+    do: Jason.decode!(body)
+
+  defp process_response!(postJson, url) do
+    images =
+      postJson["media"]
+      |> Enum.map(
+        &%{
+          url: &1["url"],
+          camo_url: Camo.Image.image_url(&1["url"])
```
Comment (liamwhite) [inline lib/philomena/scrapers/pillowfort.ex:29]: Does the API response have any preview-sized images or do you only ever get the full image?

File: lib/philomena/scrapers/pillowfort.ex [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        }
+      )
+    
+    %{
+      source_url: url,
+      author_name: postJson["username"],
+      description: postJson["title"] <> "\n----\n" <> postJson["content"],
```
Comment (liamwhite) [inline lib/philomena/scrapers/pillowfort.ex:36]: Would suggest using a [thematic break](https://github.com/chrisalley/markdown-garden/blob/master/source/guides/thematic-breaks.md) here

**Review by liamwhite — CHANGES_REQUESTED** [review]

**Comment by liamwhite** [comment]
This had no response for several months and it was basically done, so I just went ahead and finished it.

**Pull request merged**